client/app/src/Controller/Ndr/NdrController.php

```diff
@@ -229,6 +231,17 @@ public function declarationAction(Request $request, int $ndrId, S3FileUploader $
         $form = $this->createForm(FormDir\Ndr\ReportDeclarationType::class, $ndr);
         $form->handleRequest($request);
         if ($form->isSubmitted() && $form->isValid()) {
+            $validForSubmission = $ndr->validForSubmission();
+            if (count($validForSubmission['msg']) > 0) {
+                foreach ($validForSubmission['msg'] as $message) {
+                    $this->logger->warning(sprintf('Client id %s has submission issue: %s', $clientId, $message));
+                }
+
+                if (!$validForSubmission['valid']) {
+                    throw new \RuntimeException(sprintf('Submission validation failed for client id '.$clientId));
```

Comment:
Are we handling validation using exceptions? Should we be giving the user information about why their submission failed? Also, I don't think the client ID will mean much to them. (Assuming the runtime exception will cause something to be displayed to the user.)

I think this is OK short term, but we should try to give the validation errors back to the user. Perhaps put that in a bug ticket?

Reply:
Yeah no this will get swallowed by symfony so user will just see the generic application error screen. As discussed, the actual fix for the root cause of this will be further upstream before they get to this point. This simply protects against us creating the report pdf and other db updates if the app isn't in fit state to submit. I mentioned doing validation but it would really require redirects and messaging around what user needs to do etc. It's easier for us to simply throw this error and fix the upstream (another bug ticket).

In terms of what we see, this gives us a CRITICAL php error in the logs which is nice for us as we get alerted and the message will make sense to us.